lightning/src/ln/channel.rs

```diff
@@ -4217,6 +4208,48 @@ impl<Signer: Sign> Channel<Signer> {
 	}
 }
 
+fn is_unsupported_shutdown_script(their_features: &InitFeatures, script: &Script) -> bool {
+	// We restrain shutdown scripts to standards forms to avoid transactions not propagating on the p2p tx-relay network
+
+	// Providing an empty is supported
+	let script_lenght = script.len();
+	if script_lenght == 0 {
```

Comment:
This means we'll accept an empty script in `shutdown`, which we definitely cannot do - an empty script will be nonstandard. I think instead just check the 0-length case directly in the accept_channel/open_channel handling.